rust-src/bulletproofs/src/utils.rs

```diff
@@ -0,0 +1,132 @@
+use crypto_common::*;
+use crypto_common_derive::*;
+use curve_arithmetic::Curve;
+use ff::Field;
+#[cfg(test)]
+use rand::Rng;
+/// Struct containing generators G and H needed for range proofs
+#[allow(non_snake_case)]
+#[derive(Debug, Clone, Serialize, SerdeBase16Serialize)]
+pub struct Generators<C: Curve> {
+    #[size_length = 4]
+    pub G_H: Vec<(C, C)>,
+}
+
+impl<C: Curve> Generators<C> {
+    /// **Warning** do not use in production!
+    /// This **unsafely** generates a list of generators of a given size for
```

Comment:
THere is nothing inherently unsafe about this method. It is perfectly safe as long as the Rng is secure enough.

This comment is needless scaremongering.

Why does it even have to be pub if you've put it behind a test feature?

Reply:
Even if RNG is completely random, it is unsafe in the sense that it (may) generate a random exponent x and then compute g^x. In that sense, the discrete log is "known" to the computer generating it and it must be deleted safely etc. So I think it is correct to say that this function should not be used in production.

Reply:
I revised the comment and added an explanation why it should not be used in production. Is this OK?